modules/database/src/ioc/dbStatic/dbStaticLib.c

```diff
@@ -736,14 +736,17 @@ static long dbAddOnePath (DBBASE *pdbbase, const char *path, unsigned length)
 {
     ELLLIST     *ppathList;
     dbPathNode  *pdbPathNode;
+    unsigned sz;
 
     if(!pdbbase) return(-1);
     ppathList = (ELLLIST *)pdbbase->pathPvt;
 
     pdbPathNode = (dbPathNode *)dbCalloc(1, sizeof(dbPathNode));
     pdbPathNode->directory = (char *)dbCalloc(length+1, sizeof(char));
-    strncpy(pdbPathNode->directory, path, length);
-    pdbPathNode->directory[length] = '\0';
+    sz = sizeof(pdbPathNode->directory);
+    if (sz > 0)
+        strncpy(pdbPathNode->directory, path, sz);
+    pdbPathNode->directory[sz] = '\0';
     ellAdd(ppathList, &pdbPathNode->node);
```

Comment:
I think you need `length` rather than `sz`  in these lines - I believe `sz` will always be `sizeof(char*)`

Reply:
Indeed.
What is the purpose of the test? Does strncpy do something bad when the size parameter is 0?
Also, since we used calloc, there is no need to set `pdbPathNode->directory[length] = '\0';`, so this line could be removed.
What is the exact warning you get?

Reply:
Are you sure about not be necessary the last '\0' ?

The warning I'm having, on the original code, is:

```
In function ‘dbAddOnePath’,                                                                                                                                         
    inlined from ‘dbAddPath.part.7’ at ../dbStatic/dbStaticLib.c:730:16,                                                                                            
    inlined from ‘dbAddPath’ at ../dbStatic/dbStaticLib.c:663:6:                                                                                                    
../dbStatic/dbStaticLib.c:745:5: warning: ‘strncpy’ output truncated before terminating nul copying 1 byte from a string of the same length [-Wstringop-truncation] 
     strncpy(pdbPathNode->directory, path, length);                                                                                                                 
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~      
```

Reply:
I've already removed change from dbStaticLib.c, until I discover a better way to avoid this warning